src/usecases/handle-figma-file-update-event-use-case.ts

```diff
@@ -8,8 +8,29 @@ import {
 } from '../infrastructure/repositories';
 
 export const handleFigmaFileUpdateEventUseCase = {
-	execute: async (webhookId: string, fileKey: string): Promise<void> => {
+	execute: async (
+		webhookId: string,
+		fileKey: string,
+		passcode: string,
+	): Promise<void> => {
 		const figmaTeam = await figmaTeamRepository.getByWebhookId(webhookId);
+		const [connectInstallation, associatedFigmaDesigns] = await Promise.all([
```

Comment:
**suggestion:** Consider query required data only after authorizing a request to reduce a risk of a denial-of-service attack. 

In order to authorize a request, we have to call `figmaTeamRepository.getByWebhookId(webhookId)` but a potentially expensive `associatedFigmaDesignRepository.findManyByFileKeyAndConnectInstallationId` call is not required for authorization.

While a sequential execution can slightly increase the latency, it can be a safer solution and should not be an issue for a background task (which is, normally, not time-sensitive).